packer/conf/buildkite-agent/hooks/environment

```diff
@@ -4,6 +4,11 @@ set -eu -o pipefail
 # shellcheck source=/dev/null
 source ~/cfn-env
 
+# We set BUILDKITE_STACK_REGION in the cfn-env file, so if no value is set by the job
+# then we use that to set the two environment variables that aws tools depend on
+export AWS_DEFAULT_REGION="${AWS_DEFAULT_REGION:-$BUILDKITE_STACK_REGION}"
+export AWS_REGION="${AWS_REGION:-$BUILDKITE_STACK_REGION}"
```

Comment:
Would this mean that if someone changed their step config's env vars, the secrets/ecr code below this would be affected, and not just their command?

Reply:
Should we perhaps store the original value of those `AWS_` envs (in case they’re specified), then set it to the `BUILDKITE_STACK_REGION` value (this original code), then restore the `AWS_` envs to their original value after we’ve run the plugins?